src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEFieldSymbol.cs

```diff
@@ -239,13 +239,12 @@ private bool IsFixedBuffer(out int fixedSize, out TypeSymbol fixedElementType)
             if (containingPEModule.Module.HasFixedBufferAttribute(_handle, out elementTypeName, out bufferSize))
             {
                 var decoder = new MetadataDecoder(containingPEModule);
-                var specialTypeName = MetadataHelpers.DecodeTypeName(elementTypeName).TopLevelType;
-                var specialType = SpecialTypes.GetTypeFromMetadataName(specialTypeName);
-                if (specialType != SpecialType.None &&
-                    specialType.FixedBufferElementSizeInBytes() != 0)
+                var elementType = decoder.GetTypeSymbolForSerializedType(elementTypeName);
+                Debug.Assert(!elementType.IsErrorType());
```

Comment:
The assert is too optimistic. The type could be missing if reference is not given to the compilation. Consider removing.